File: Spring/ITraining_API/src/main/java/Entity/UserAccount.java
package Entity;

import jakarta.persistence.*;
import org.hibernate.annotations.GenericGenerator;

@Entity
public class UserAccount {

    @Id
    @GeneratedValue(strategy= GenerationType.AUTO,generator="native")
    @GenericGenerator(name = "native",strategy = "native")
    @Column(name = "user_id")
    private int id;

    private String firstName;

    private String lastName;

    private String email;

    private String phone;

    private String password;


}

File: Angular/itraining_front/src/model/userAccount.ts
export interface IUserAccount {
  id: number;
  firstName: string;
  lastName: string;
  email: string;
  phone: string;
  password: string;
}

export interface ITeacherAccount extends IUserAccount {
  subject: string;
  experience: number;
}


export interface IAdministratorAccount extends IUserAccount {}


export interface ILearnerAccount extends IUserAccount {}
